### DISCUSSION TITLE
How can I copy the source block?

### DISCUSSION THREAD

**Discussion by madorian** [karthink/gptel#388] [General]
Is there a command to copy the source block to the kill-ring?

I think that would save me a few hours every day;)

**Comment by karthink** [comment]
That would be an Org/Markdown feature I think.

```elisp
(defun org-babel-kill-ring-save ()
  (interactive)
  (let* ((elem (org-element-context))
         (start (progn (goto-char (org-element-begin elem))
                       (forward-line) (point)))
         (end (progn (goto-char (org-element-end elem))
                     (skip-chars-backward " \n")
                     (forward-line 0)
                     (point))))
    (unless (= start end) (kill-ring-save start end))))
```

If the source block is intended for another Emacs buffer, using a chat buffer and copying output afterwards is probably an inefficient/tiresome way to go about this.  You can redirect the LLM response instead when making the request.  Just be sure to add a directive to produce only code as a response.

![image](https://github.com/user-attachments/assets/d8042197-4e7e-4967-b5ad-dacf123c79a6)
